lib/collection/src/collection/resharding.rs

```diff
@@ -162,30 +181,31 @@ impl Collection {
         let _ = self.stop_resharding_task(&resharding_key).await;
         shard_holder.finish_resharding_unchecked(&resharding_key)?;
 
-        // Update the stored shard count, ensures we load the new shard on restart
+        Ok(())
+    }
+
+    pub async fn abort_resharding(&self, resharding_key: ReshardKey) -> CollectionResult<()> {
+        let _ = self.stop_resharding_task(&resharding_key).await;
+
+        // Decrease the persisted shard count, ensures we don't load dropped shard on restart
         {
             let mut config = self.collection_config.write().await;
-            config.params.shard_number = config.params.shard_number.saturating_add(1);
 
-            // If using automatic sharding, the reshard key shard ID must correspond with number
+            #[cfg(debug_assertions)]
             if config.params.sharding_method.unwrap_or_default() == ShardingMethod::Auto {
-                debug_assert_eq!(
+                assert_eq!(
                     config.params.shard_number.get() - 1,
-                    resharding_key.shard_id,
+                    resharding_key.shard_id
                 );
             }
```

Comment:
I'd actually leave it as it was before in `finish_resharding`. E.g., just `debug_assert_eq` under conditional, without explicit `#[cfg(debug_assertions)]`. I'd expect the compiler to optimize the conditional away in release.